File: Extensions/Tessa.Extensions.Client/Cards/PnrIncomingStoreExtension.cs
using System;
using System.Threading.Tasks;
using Tessa.Cards;
using Tessa.Cards.Extensions;
using Tessa.UI;

namespace Tessa.Extensions.Client.Cards
{
    class PnrIncomingStoreExtension : CardStoreExtension
    {
        #region Constructors and Fields

        public PnrIncomingStoreExtension(IUIHost uiHost)
        {
            this.uiHost = uiHost;
        }

        private readonly IUIHost uiHost;
        
        #endregion

        #region Base overrides

        public override async Task AfterRequest(ICardStoreExtensionContext context)
        {
            var card = context.Request.TryGetCard();
            if (card == null || card.StoreMode != CardStoreMode.Insert)
            {
                return;
            }

            var getInfo = context.Response.Info.TryGetValue("similarIncominGuid", out object similarObj);
            if (getInfo && similarObj != null)
            {
                Guid similarCardID = (Guid) similarObj;
                bool? result = TessaDialog.ConfirmWithCancel(
                    "В системе уже имеется карточка с похожими данными. Желаете ее открыть?", "Найден аналогичный документ");
                if (!result.HasValue)
                {
                    return;
                }
                else
                {
                    if (result.Value)
                    {
                        await this.uiHost.OpenCardAsync(similarCardID);
                    }
                }
            }
        }
        #endregion
    }
}

File: WebClient SDK/src/PnrScripts/Cards/PnrIncomingStoreExtension.ts
import { CardStoreExtension, ICardStoreExtensionContext } from "tessa/cards/extensions";
import { showConfirmWithCancel } from "tessa/ui";
import { PnrCardTypes } from "../Shared/PnrCardTypes";
import { openCard } from 'tessa/ui/uiHost';
import { Guid } from "tessa/platform";

export class PnrIncomingStoreExtension extends CardStoreExtension {
    public async afterRequest(context: ICardStoreExtensionContext) {
        if (!context) {
            return;
        }
        if (!context.cardType
            || !(Guid.equals(context.cardType.id, PnrCardTypes.PnrIncomingTypeID))
            || context.response == null) {
            return;
        }

        let similarCardID = context.response.info['similarIncominGuid'].$value;
        const dialogResult = await showConfirmWithCancel(
            'В системе уже имеется карточка с похожими данными. Желаете ее открыть?',
            'Найден аналогичный документ');
        if (dialogResult !== null || dialogResult !== false) {
            await openCard({cardId: similarCardID});
        }
    }
}
